cfn-resources/cloud-outage-simulation/cmd/resource/resource.go

```diff
@@ -0,0 +1,344 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"github.com/spf13/cast"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+const (
+	Simulating        = "SIMULATING"
+	Starting          = "STARTING"
+	StartingRequested = "START_REQUESTED"
+	Complete          = "COMPLETE"
+)
+
+var RequiredFields = []string{constants.ClusterName, constants.ProjectID}
+var SimulationStatus = []string{Simulating, Starting, StartingRequested}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+
+	// Validate required fields in the request
+	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == constants.EmptyString {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, pe := util.NewAtlasClient(&req, currentModel.Profile)
+	if pe != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
+		return *pe, nil
+	}
+
+	if req.CallbackContext != nil {
+		return validateProgress(atlas, currentModel, Simulating)
```

Comment:
Oh I see you set the success here.This is confusing since the name suggests that this should return only an error if the validation failed. Can we move the handler.ProgressEvent in this function?

Reply:
validateProgress method will validates status until it returns simulation success or failure and Respond with corresponding ProgressEvent.